vulnfeeds/cpp/main.go

```diff
@@ -75,84 +76,82 @@ func InScopeGitRepo(repoURL string) bool {
 	return true
 }
 
-func GetRepoDir(repoURL string, repobasedir string) string {
-	parsedURL, err := url.Parse(repoURL)
+// Take an unnormalized version string, a repo, the pre-normalized mapping of tags to commits and return a GitCommit.
+func GitVersionToCommit(version string, repo string, normalizedTags map[string]git.NormalizedTag) (gc cves.GitCommit, e error) {
+	normalizedVersion, err := cves.NormalizeVersion(version)
 	if err != nil {
-		return ""
+		return gc, err
 	}
-	return path.Join(repobasedir, parsedURL.Hostname(), parsedURL.Path)
-}
-
-// Either clones the repository or returns the one previously cloned.
-func GetRepo(repoURL string, repobasedir string) (r *git.Repository, e error) {
-	repodir := GetRepoDir(repoURL, repobasedir)
-	// TODO: reorder to open, then clone, once behaviour and failure modes understood
-	Logger.Infof("Cloning %s", repoURL)
-	r, err := git.PlainClone(repodir, false, &git.CloneOptions{
-		URL: repoURL,
-	})
-	if err != nil && err != git.ErrRepositoryAlreadyExists {
-		return nil, err
+	// Try a straight out match first.
+	// TODO try fuzzy prefix matches also.
+	normalizedTag, ok := normalizedTags[normalizedVersion]
+	if !ok {
+		return gc, fmt.Errorf("Failed to find a commit for version %q normalized as %q", version, normalizedVersion)
 	}
-	r, err = git.PlainOpen(repodir)
-	if err != nil {
-		return nil, err
-	}
-	return r, nil
+	return cves.GitCommit{
+		Repo:   repo,
+		Commit: normalizedTag.Commit,
+	}, nil
 }
 
-// Examines repos and tries to convert version tags to commits
-func GitVersionToCommit(versions cves.VersionInfo, repos []string, repobasedir string) (v cves.VersionInfo, e error) {
+// Examines repos and tries to convert versions to commits by treating them as Git tags.
+// Takes a CVE ID string (for logging), cves.VersionInfo with AffectedVersions and
+// no FixCommits and attempts to add FixCommits.
+func GitVersionsToCommit(CVE string, versions cves.VersionInfo, repos []string, cache git.RepoTagsCache) (v cves.VersionInfo, e error) {
 	// versions is a VersionInfo with AffectedVersions and no FixCommits
 	// v is a VersionInfo with FixCommits included
 	v = versions
 	for _, repo := range repos {
-		r, err := GetRepo(repo, repobasedir)
+		normalizedTags, err := git.NormalizeRepoTags(repo, cache)
 		if err != nil {
-			Logger.Warnf("Failed to get %s: %v", repo, err)
+			Logger.Warnf("[%s]: Failed to normalize tags for %s: %v", CVE, repo, err)
 			continue
 		}
 		for _, av := range versions.AffectedVersions {
 			// We can't map a non-existent version to a tag and commit, so skip.
-			if av.Fixed == "" {
+			if av.Fixed == "" && av.LastAffected == "" {
```

Comment:
nit: we can probably just skip this condition completely since it's redundant.

Reply:
Removed in [993ea85](https://github.com/google/osv.dev/pull/1105/commits/993ea8584d7e429f00b8b358c25d33815fdf4d2b)